lib/src/native/realm_core.dart

```diff
@@ -2009,25 +2019,54 @@ extension on realm_object_id {
   }
 }
 
-// Helper enum for converting Level
-enum _LogLevel {
-  all(RealmLogLevel.all),
-  trace(RealmLogLevel.trace),
-  debug(RealmLogLevel.debug),
-  detail(RealmLogLevel.detail),
-  info(RealmLogLevel.info),
-  warn(RealmLogLevel.warn),
-  error(RealmLogLevel.error),
-  fatal(RealmLogLevel.fatal),
-  off(RealmLogLevel.off);
-
-  final Level loggerLevel;
-  const _LogLevel(this.loggerLevel);
-
-  factory _LogLevel.fromLevel(Level level) {
-    for (final candidate in _LogLevel.values) {
-      if (level.value > candidate.loggerLevel.value) return candidate;
+extension LevelExt on Level {
+  int toInt() {
+    if (this == Level.ALL) {
+      return 0;
+    } else if (name == "TRACE") {
+      return 1;
+    } else if (name == "DEBUG") {
+      return 2;
+    } else if (name == "DETAIL") {
+      return 3;
+    } else if (this == Level.INFO) {
+      return 4;
+    } else if (this == Level.WARNING) {
+      return 5;
+    } else if (name == "ERROR") {
+      return 6;
+    } else if (name == "FATAL") {
+      return 7;
+    } else if (this == Level.OFF) {
+      return 8;
+    } else {
+      // if unknown logging is off
+      return 8;
     }
-    return _LogLevel.off;
+  }
+
+  static Level fromInt(int value) {
+    if (value == 0) {
```

Comment:
I find massive if-statements like this very hard to read - can we rework this and the above to be a switch statement?

Reply:
yes I guess I could do that.